extensions/amp-next-page/amp-next-page.md

```diff
@@ -25,237 +25,272 @@ limitations under the License.
+
+The following configuration returned from the server will recommend two more documents for the user to read and tell `amp-next-page` to query `https://example.com/more-pages` when the user finishes reading the provided recommendations.
+
+```json
+{
+  "pages": [
     {
-      "requests": {
-        "nextpage": "https://foo.com/pixel?RANDOM&toURL=${toURL}"
-      },
-      "triggers": {
-        "trackScrollThrough": {
-          "on": "amp-next-page-scroll",
-          "request": "nextpage"
-        },
-        "trackClickThrough": {
-          "on": "amp-next-page-click",
-          "request": "nextpage"
-        }
-      }
+      "image": "https://example.com/image1.jpg",
+      "title": "This article shows first",
+      "url": "https://example.com/article1.amp.html"
+    },
+    {
+      "image": "https://example.com/image2.jpg",
+      "title": "This article shows second",
+      "url": "https://example.com/article2.amp.html"
     }
-  </script>
-</amp-analytics>
+  ],
+  "next": "https://example.com/more-pages"
+}
+```
+
+##### URL substitutions
+
+The `amp-next-page` `src` allows all standard URL variable substitutions. See the [Substitutions Guide](../../spec/amp-var-substitutions.md) for more info. For example:
+
+```html
+<amp-next-page src="https://foo.com/config.json?RANDOM"></amp-next-page>
 ```
 
-### Recommendation box
+This URL may make a request to something like `https://foo.com/config.json?0.8390278471201` where the RANDOM value is randomly generated upon each impression.
+
+### Footer recommendation box
 
-If the user reaches the end of a page before the next has loaded (or if the
-next page fails to load), a box will be displayed with links to the next three
-pages. This box will also be displayed after the maximum number of articles
-have been rendered.
+The `<amp-next-page>` component renders the footer recommendation box if one of the following situations arise:
 
-The appearance of these links can be customized by styling the following
-classes:
+- The user reaches the end of the page before the next page had loaded.
+- The next page fails to load.
+- If the `max-pages` attribute is specified and the number of displayed pages is met.
 
-- `.amp-next-page-links` for the containing element
-- `.amp-next-page-link` for an individual link
-- `.amp-next-page-image` for the link image
-- `.amp-next-page-text` for the link text
+The footer recommendation box contains links to the remaining pages. The default footer recommendation box renders the specified image and title used in the JSON configuration and can be styled as specified in the Styling section.
+
+The footer recommendation box can also be provided as a custom component inside `<amp-next-page>` as any element that has the `footer`. It can also be templated via `amp-mustache` or other templating engines, in which case it will be passed an array of the remaining `pages`, each being an object with a `title`, `url` and `image`. Example:
```

Comment:
I'm sorry but I do not understand any of this.

How is the footer recommendation box provided as a custom component? What in the following code sample signifies that this is the footer recommendation box? 

How does including the custom component inside `<amp-next-page>` effect the default behavior?